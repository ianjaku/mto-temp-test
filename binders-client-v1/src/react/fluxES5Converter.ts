/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
