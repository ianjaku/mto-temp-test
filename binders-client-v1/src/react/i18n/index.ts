import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
