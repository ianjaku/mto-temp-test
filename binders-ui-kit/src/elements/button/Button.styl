@require "../../vars.styl";
@require "../../style/global.styl";

makeButtonDisabled()
    opacity: 0.2
    cursor: default

.button
    font-family: defaultFontName, sans-serif
    color: whiteColor
    background-color: baseColor
    padding: 10px 34px
    border-radius: 5px
    display: inline-flex
    align-items: center
    justify-content: center
    text-align: center
    transition: all 0.2s ease-in-out
    font-size: 13px
    font-weight: 500
    margin: 5px 0 5px 0
    position: relative
    text-decoration: none
    line-height: 1em
    border: 1px solid baseColor
    @media rwd.tabletAndAbove
        padding: 10px 40px
    &:not(.button--disabled)&:not(.button--unhoverable)
        cursor: pointer
        &:hover
            background-color: accentColor
            border-color: accentColor
    &:focus
        outline: none
    &--disabled
        makeButtonDisabled()
    &--secondary
        background-color: transparent
        border-color: baseColor
        color: baseColor
        &.branded
            border-color: var(--mainColor)
            color: var(--mainColor)
        &:not(.button--disabled)&:not(.button--unhoverable)
            &:hover
                border-color: baseColor
                background-color: baseColor
                color: white
                &.branded
                    border-color: var(--mainColor-hover)
                    background-color: var(--mainColor-hover)
        &^[0]--disabled
            makeButtonDisabled()
    &--CTA
        background-color: accentColor
        border-color: accentColor
        &.branded
            background-color: var(--mainColor)
            border-color: var(--mainColor)
        &:not(.button--disabled)&:not(.button--unhoverable)
            &:hover
                box-shadow: 0px 2px 3px 0px rgba(0,0,0,0.35)
                &.branded
                    background-color: var(--mainColor)
                    border-color: var(--mainColor)
        &^[0]--disabled
            makeButtonDisabled()
    &--borderless
        border: none
    &-round
        width: 24px;
        height: 24px;
        border-radius: 15px;
        background-color: accentColor;
        color: whiteColor;
        border: none;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
        cursor: pointer;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        img
            box-shadow: none
        &:focus
            outline: none;

        .material-icons
            font-size: 20px !important;
            color: whiteColor !important;
    &-loader
        right: 10px
        top: 8px
        color: whiteColor !important
        &--secondary
            color: baseColor !important

    &-link
        position: absolute
        display: block
        top: 0
        bottom: 0
        right: 0
        left: 0

