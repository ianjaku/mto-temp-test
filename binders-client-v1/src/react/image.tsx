/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
