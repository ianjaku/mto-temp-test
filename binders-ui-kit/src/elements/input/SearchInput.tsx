import * as React from "react";
import ThemedInput, { IInputProps } from "./index";
import SearchIcon from "../icons/Search";
import cx from "classnames";
import "./input.styl";

const SearchInput: React.FC<Omit<IInputProps, "ref">> = (props) => {
    return (
        <div className={cx("input-wrapper input-wrapper-search", { "input-wrapper--inverted": props.inverted })}>
            <span className="input-icon input-icon--center">
                <SearchIcon />
            </span>
            <ThemedInput {...props} type="text" />
        </div>
    );
}

export default SearchInput;
