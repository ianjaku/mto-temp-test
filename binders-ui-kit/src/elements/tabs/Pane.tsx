import * as React from "react";

export interface PaneProps {
    label: string;
    onClick?: () => boolean | void;
    testId?: string;
}

export const Pane: React.FC<PaneProps> = (props) => {
    return (
        <div className="tabs-pane">
            {props.children}
        </div>
    );
};
