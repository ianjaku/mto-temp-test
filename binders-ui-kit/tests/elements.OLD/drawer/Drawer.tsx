import * as React from "react";
import Button from "../../../src/elements/button";
import Drawer from "../../../src/elements/drawer";
import Pane from "../../../src/elements/rightPane";
import PaneItem from "../../../src/elements/rightPane/item";
import SearchInput from "../../../src/elements/input/SearchInput";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

function onClick() {
    alert("button clicked!");
}

function getDrawer() {
    return wrapWithTheme(
        <Pane ribbonsTopHeight={0}>
            <PaneItem iconName={"add"} >
                <Drawer title="Media1">
                    <Button onClick={onClick} text="Horses" secondary={true} />
                </Drawer>
            </PaneItem>
            <PaneItem iconName={"image"} >
                <Drawer title="Media2">
                    <Button onClick={onClick} text="Are" secondary={true} />
                </Drawer>
            </PaneItem>
            <PaneItem iconName={"settings"} >
                <Drawer title="Media3">
                    <Button onClick={onClick} text="Awesome" secondary={true} />
                </Drawer>
            </PaneItem>
            <PaneItem iconName={"share"} >
                <Drawer title="Media4">
                    <SearchInput inverted={true} />
                </Drawer>
            </PaneItem>
        </Pane>,
    );

}
let drawer = null;

describe("Drawer", () => {

    beforeEach(() => {
        drawer = getDrawer();
    });


    test("drawer (snapshot)", () => {
        const modal = create(drawer);
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("click on icon (mount: enzyme+jsdom)", () => {
        const element = mount(drawer);
        element.find("button")
            .at(0).simulate("click");

        const dr = element.find(".drawer.is-opened");
        expect(dr.length).not.toBe(0);
    });

    test("click on icon twice (mount: enzyme+jsdom)", () => {
        const element = mount(drawer);
        element.find("button")
            .at(0).simulate("click");

        element.find("button")
            .at(0).simulate("click");

        const dr = element.find(".drawer.is-opened");
        expect(dr.length).toBe(0);
    });

    test("close drawer by x (mount: enzyme+jsdom)", () => {
        const element = mount(drawer);
        element.find("button")
            .at(0).simulate("click");

        element.find("button.drawer-container-header-icon")
            .at(0).simulate("click");

        const dr = element.find(".drawer.is-opened");
        expect(dr.length).toBe(0);
    });

    test("check if icon is active (mount: enzyme+jsdom)", () => {
        const element = mount(drawer);
        element.find(".rightPane button")
            .at(0).simulate("click");

        const button = element.find(".pane-item-icon").at(0);
        expect(button.hasClass("pane-item-icon--active"));
    });
});
