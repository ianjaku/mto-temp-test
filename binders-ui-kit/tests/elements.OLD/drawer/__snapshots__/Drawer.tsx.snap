// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer drawer (snapshot) 1`] = `
<div
  className="rightPane"
>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": "40px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": "5px",
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "40px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="pane-item-icon material-icons"
          color="white"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "white",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          add
        </span>
      </div>
    </button>
    <div
      className="drawer"
      style={undefined}
    >
      <div
        className="drawer-container"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#484744",
            "borderRadius": "0px",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                   0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "white",
            "fontFamily": "Roboto, sans-serif",
            "height": "100%",
            "left": "auto",
            "muiPrepared": true,
            "overflow": "auto",
            "padding": "32px 16px",
            "position": "fixed",
            "right": 0,
            "top": 0,
            "transform": "translate(410px, 0)",
            "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 400,
            "zIndex": 1300,
          }
        }
      >
        <header
          className="drawer-container-header"
        >
          <span
            className="drawer-container-header-title"
          >
            Media1
          </span>
          <button
            className="drawer-container-header-icon"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "marginRight": "-10px",
                "marginTop": "-10px",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                className="material-icons "
                color="white"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "white",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              >
                close
              </span>
            </div>
          </button>
        </header>
        <div
          className="drawer-container-body"
        >
          <div
            className="button button--secondary"
            onBlur={undefined}
            onClick={[Function]}
            tabIndex={undefined}
          >
            Horses
            
            
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": "40px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": "5px",
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "40px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="pane-item-icon material-icons"
          color="white"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "white",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          image
        </span>
      </div>
    </button>
    <div
      className="drawer"
      style={undefined}
    >
      <div
        className="drawer-container"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#484744",
            "borderRadius": "0px",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                   0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "white",
            "fontFamily": "Roboto, sans-serif",
            "height": "100%",
            "left": "auto",
            "muiPrepared": true,
            "overflow": "auto",
            "padding": "32px 16px",
            "position": "fixed",
            "right": 0,
            "top": 0,
            "transform": "translate(410px, 0)",
            "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 400,
            "zIndex": 1300,
          }
        }
      >
        <header
          className="drawer-container-header"
        >
          <span
            className="drawer-container-header-title"
          >
            Media2
          </span>
          <button
            className="drawer-container-header-icon"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "marginRight": "-10px",
                "marginTop": "-10px",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                className="material-icons "
                color="white"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "white",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              >
                close
              </span>
            </div>
          </button>
        </header>
        <div
          className="drawer-container-body"
        >
          <div
            className="button button--secondary"
            onBlur={undefined}
            onClick={[Function]}
            tabIndex={undefined}
          >
            Are
            
            
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": "40px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": "5px",
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "40px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="pane-item-icon material-icons"
          color="white"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "white",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          settings
        </span>
      </div>
    </button>
    <div
      className="drawer"
      style={undefined}
    >
      <div
        className="drawer-container"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#484744",
            "borderRadius": "0px",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                   0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "white",
            "fontFamily": "Roboto, sans-serif",
            "height": "100%",
            "left": "auto",
            "muiPrepared": true,
            "overflow": "auto",
            "padding": "32px 16px",
            "position": "fixed",
            "right": 0,
            "top": 0,
            "transform": "translate(410px, 0)",
            "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 400,
            "zIndex": 1300,
          }
        }
      >
        <header
          className="drawer-container-header"
        >
          <span
            className="drawer-container-header-title"
          >
            Media3
          </span>
          <button
            className="drawer-container-header-icon"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "marginRight": "-10px",
                "marginTop": "-10px",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                className="material-icons "
                color="white"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "white",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              >
                close
              </span>
            </div>
          </button>
        </header>
        <div
          className="drawer-container-body"
        >
          <div
            className="button button--secondary"
            onBlur={undefined}
            onClick={[Function]}
            tabIndex={undefined}
          >
            Awesome
            
            
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": "40px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": "5px",
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "40px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="pane-item-icon material-icons"
          color="white"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "white",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          share
        </span>
      </div>
    </button>
    <div
      className="drawer"
      style={undefined}
    >
      <div
        className="drawer-container"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#484744",
            "borderRadius": "0px",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                   0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "white",
            "fontFamily": "Roboto, sans-serif",
            "height": "100%",
            "left": "auto",
            "muiPrepared": true,
            "overflow": "auto",
            "padding": "32px 16px",
            "position": "fixed",
            "right": 0,
            "top": 0,
            "transform": "translate(410px, 0)",
            "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 400,
            "zIndex": 1300,
          }
        }
      >
        <header
          className="drawer-container-header"
        >
          <span
            className="drawer-container-header-title"
          >
            Media4
          </span>
          <button
            className="drawer-container-header-icon"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "marginRight": "-10px",
                "marginTop": "-10px",
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <span
                className="material-icons "
                color="white"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "white",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              >
                close
              </span>
            </div>
          </button>
        </header>
        <div
          className="drawer-container-body"
        >
          <div
            className="input-wrapper input-wrapper-search input-wrapper--inverted"
          >
            <span
              className="input-icon"
            >
              <span
                className="material-icons "
                color=""
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fontSize": 24,
                    "muiPrepared": true,
                    "position": "relative",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                  }
                }
              >
                search
              </span>
            </span>
            <input
              className="input input--inverted"
              onChange={[Function]}
              style={
                Object {
                  "width": "undefinedpx",
                }
              }
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
